### ISSUE TITLE
Poor display of Grid widget

### ISSUE THREAD

**Issue by slightlynybbled** [jarvisteach/appJar#152]
The Grid widget shows up poorly

![window](https://cloud.githubusercontent.com/assets/2328302/25241717/fcd7795c-25c4-11e7-957a-8fff413f9e9f.png)

 * always has scroll bars, even when not necessary
 * entry widgets don't stretch to the width of the field
 * overall window doesn't stretch to fit the grid, meaning that scrolling is always required.

Very nice concept for the grid widget!  Can't wait to see it working well!  Once this is complete, I will probably move from tkinter completely.

**Comment by jarvisteach** [comment]
Entry widgets do look rubbish - should be an easy fix.
Getting the container for the grid to fill its container should also be a straight-forward fix.

Managing the scroll panel is more difficult, and will take a bit more work - it should be using hide/show scrollbars that only appear as necessary, will need to investigate...

**Comment by jarvisteach** [comment]
If you manually set the size of the GUI window - you can have it not need the horizontal scroll bars.

The width of the GUI is set off the requested width of the widgets. The scroll pane (containing the grid) doesn't request a width, but rather takes what it's given - in the above example it has matched itself to the width of the previous widgets.

The scrollbars are always there, but are inactive when not needed, as you add rows the horizontal scrollbar will come into play...

**Comment by jarvisteach** [comment]
If you let me know what you;re trying to achieve - I can look at adjusting the widget, to account for the use case...

**Comment by slightlynybbled** [comment]
@jarvisteach Thank you for looking at this!  I love that tkinter is included with the standard python library, but it sometimes drives me nuts.

I placed some comments is #151 to clarify my bugs and use case.

Thanks again, brilliant work!

**Comment by jarvisteach** [comment]
Had a play with yours, it offers different functionality to mine - mine is a grid of Labels to show data, with the bottom row being Entries. By default, it is designed to just show a grid of data - with selectable Labels. Calling `.getGridSelectedCells(title)` will return all the selected cells. If you set `action`. If you set `action` you get a button at the end of each row - allowing you to *select* a row. If you then also set `addRow` you get the extra row of Entries, with their own *Submit* button...

It was built for a fairly specific use case, a long time ago!

Would you be open to me including your grid in appJar as an alternative to the existing one?

**Comment by jarvisteach** [comment]
Actually, upon closer inspection, it seems your grid does both Labels & Entries! Even better!

**Comment by slightlynybbled** [comment]
No problem at all.  I wouldn't have shared it if it weren't shareable.  I believe that the docstrings are reasonably adequate.  Let me know if you have any questions.

I often need a display grid and a separate entry grid as opposed to a single entity containing all both.  I toyed with the idea of disabling the entries to make a display grid at one point, but decided to go another direction, as you see.  Enjoy!

**Comment by slightlynybbled** [comment]
I uploaded my [tk_tools](https://github.com/slightlynybbled/tk_tools) that I normally use to github for reference.  The file that I sent to you is something of a snapshot, but not one to be trusted (I think I ended up finding an error in LabelGrid() since I sent that to you).  These are just a few widgets that I have hacked together over some time and decided to put them in a nicer format for generic consumption.  Feel free to take any of those.  I'll do a `pip install` version soon.

**Comment by jarvisteach** [comment]
Thank you - I'm back at school now, preparing the kids for their end of year exams, so don't have much time for adding new features. But I'll definitely incorporate some of these into the library - I especially like the graphing widget as an alternative to matplotlib!

**Comment by jarvisteach** [comment]
Will raise new issue regarding scroll panes and sizing.
Will rename this issue to incorporate new widgets.
Other bits have been fixed.
